### DISCUSSION TITLE
[Art competition] Static Analysis

### DISCUSSION THREAD

**Discussion by greenmonn** [prosyslab-classroom/cs524-program-analysis#274] [Art competition]
![20221201_034400](https://user-images.githubusercontent.com/17661005/204882772-e071affd-4122-417c-8fa7-903cdad558f3.jpg)

## Short Description
For me, [*Static Analysis*] means [*A monster trapped in a glass jail and people taking pictures of the monster with their own cameras (digital art)*]".

## Long Description (up to 3 sentences)
A newly implemented program is analogous to a freshly born monster with unknown hazards. We do not let the monster (program) dynamically move (execute), and try to capture the possible vulnerabilities using our own cameras with different resolutions and filters (static analyzers with different precision levels).

**Comment by KihongHeo** [comment]
Fancy artwork.
BTW, can cameras capture the deep semantics of such monsters? (한 길 사람 속은?)

**Comment by greenmonn** [comment]
Maybe, at least, we would be able to infer the deep semantics based on the more advanced observations of the poor monster (..using IR, X-ray cameras?). Although 한 길 사람 속 might be harder to capture than 열 길 program semantics, if we spend more time and memory to better know a person, I believe we can obtain a precise image of the one.

**Comment by KihongHeo** [comment]
Good.

BTW, could you resize your image to 1024x1024 to fit in with [this gallery](https://github.com/prosyslab-classroom/cs524-program-analysis/blob/main/hof.md)?

**Comment by greenmonn** [comment]
Sure! I substituted the original collage to the resized one.

**Comment by KihongHeo** [comment]
Thanks. It seems that you really like green monsters (the drawing and your GH id).
